feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c

```diff
@@ -980,119 +1006,172 @@ void vif_hs20_update(struct schema_Hotspot20_Config *hs2conf)
 	}
 }
 
+/* Mesh options table */
+#define VIF_SCHEMA_MESH_OPT_SZ            32
+#define VIF_SCHEMA_MESH_OPTS_MAX          2
+
+const char vif_mesh_options_table[VIF_SCHEMA_MESH_OPTS_MAX][VIF_SCHEMA_MESH_OPT_SZ] =
+{
+		SCHEMA_CONSTS_MESH_MCAST_RATE,
+		SCHEMA_CONSTS_MESH_MTU,
+};
+
+static void vif_mesh_opt_set(struct blob_buf *b, struct blob_buf *b1,
+		const struct schema_Wifi_VIF_Config *vconf)
+{
+	int i;
+	const char *opt;
+	const char *val;
+
+	for (i = 0; i < VIF_SCHEMA_MESH_OPTS_MAX; i++) {
+		opt = vif_mesh_options_table[i];
+		val = SCHEMA_KEY_VAL_NULL(vconf->mesh_options, opt);
+
+		if (strcmp(opt, "mcast_rate") == 0) {
+			if (!val)
+				blobmsg_add_u32(b, "mcast_rate", 24000);
+			else
+				blobmsg_add_u32(b, "mcast_rate", atoi(val));
+		}
+		else if (strcmp(opt, "mtu") == 0) {
+			if (!val)
+				blobmsg_add_u32(b1, "mtu", 1532);
+			else
+				blobmsg_add_u32(b1, "mtu", atoi(val));
+		}
+	}
+}
+
 bool target_vif_config_set2(const struct schema_Wifi_VIF_Config *vconf,
-			    const struct schema_Wifi_Radio_Config *rconf,
-			    const struct schema_Wifi_Credential_Config *cconfs,
```

Comment:
lots of code reformatting in this patch making it hard to see what has actually changed. could you please remove the reformatting part ?

Reply:
I think the whitespace change is to due to the mesh or ap decision resulting in the AP specific parameters adding 1 level of TAB to the whitespace.  The recommendation to separate the mesh and ap portions into two separate helper functions is probably the solution here.

Reply:
Yeah. Already done with the changes. Currently resolving the merge conflicts with your latest commit. New patch should be pushed in a while.

Reply:
This one was badly indented. So had to change.